src/super_gradients/training/metrics/depth_estimation_metrics.py

```diff
@@ -0,0 +1,250 @@
+        self.apply_sigmoid = apply_sigmoid
+
+    def process_preds_and_target(self, preds: Union[Tensor, Sequence[Tensor]], target: Tensor) -> Tuple[Tensor, Tensor]:
+        """
+        Process predictions and target tensors for depth estimation metrics:
+        - If a sequence is returned by the model -> sets preds to the first element
+        - Squeezes the dummy dimension (i.e dim1) when preds.ndim == 4
+        - Applies sigmoid to preds if apply_sigmoid is True
+        - Removes entries to ignore where ignore_val is present in target
+
+        :param preds: Model predictions, either a tensor or a sequence of tensors.
+        :param target: Ground truth depth map.
+        :return: Processed predictions and target tensors.
+        """
+
+        if isinstance(preds, Sequence):
+            preds = preds[0]
+
+        if preds.ndim == 4:
+            preds = preds.squeeze(1)
+
+        if self.apply_sigmoid:
+            preds = torch.sigmoid(preds)
+
+        if self.ignore_val is not None:
+            non_ignored = preds != self.ignore_val
+            preds = preds[non_ignored]
+            target = target[non_ignored]
+
+        return preds, target
+
+
+@register_metric(Metrics.MeanSquaredErrorWithIgnored)
+class MeanSquaredErrorWithIgnored(MeanSquaredError, DepthEstimationMetricMixin):
+    """MeanAbsoluteError, MeanAbsolutePercentageError
+    Mean Squared Error metric for depth estimation with support for ignored values.
+
+    :param squared: If True returns MSE value, if False returns RMSE value.
+    :param ignore_val: Value to be ignored when computing the metric.
+    :param apply_sigmoid: Whether to apply the sigmoid function to predictions before updating the metric.
+    """
+
+    def __init__(self, squared: bool = True, ignore_val: Optional[float] = None, apply_sigmoid: bool = False):
+        super().__init__(squared=squared)
+        DepthEstimationMetricMixin.__init__(self, ignore_val, apply_sigmoid)
+
+    def update(self, preds: Tensor, target: Tensor) -> None:
+        """
+        Update the metric with model predictions and target depth map.
+
+        :param preds: Model predictions.
+        :param target: Ground truth depth map.
+        """
+        preds, target = self.process_preds_and_target(preds, target)
+        super().update(preds, target)
+
+
+@register_metric(Metrics.MSE)
+class MSE(MeanSquaredErrorWithIgnored):
+    """
+    Mean Squared Error metric (squared) for depth estimation with support for ignored values.
+
+    :param ignore_val: Value to be ignored when computing the metric.
+    :param apply_sigmoid: Whether to apply the sigmoid function to predictions before updating the metric.
+    """
+
+    def __init__(self, ignore_val: Optional[float] = None, apply_sigmoid: bool = False):
+        super().__init__(squared=True, ignore_val=ignore_val, apply_sigmoid=apply_sigmoid)
+
+
+@register_metric(Metrics.RMSE)
+class RMSE(MeanSquaredErrorWithIgnored):
```

Comment:
This class is the same as MSE. Base class inheritance is wrong?

Reply:
Accidentally left default value for squared argument which is the difference. Fixed.